tests/installation/setup_online_repos.pm

```diff
@@ -20,6 +20,16 @@ sub run() {
     # ordered according to real repos lists
     my @default_repos = qw(update-non-oss update-oss main-non-oss main-oss debug-main untested-update debug-update source);
 
+    if (get_var('VERSION', '') !~ /42\.[1-2]/) {
```

Comment:
Depends on what exactly should be tested here, this way if the wrong workflow is somehow triggered the test fails.
If that is not important I can of course replace it with the detection mased on the needle tag.